onnxruntime/core/optimizer/pre_shape_node_elimination.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "core/optimizer/rewrite_rule.h"
+
+namespace onnxruntime {
+
+/**
+@class PreShapeNodeElimination
+
+Rewrite rule that eliminates some particular nodes if the next node is a Shape node.
+
+It is attempted to be triggered only on nodes with op type "Cast".
+*/
+class PreShapeNodeElimination : public RewriteRule {
+ public:
+  PreShapeNodeElimination() noexcept : RewriteRule("PreShapeNodeElimination") {}
+
+  std::vector<std::string> TargetOpTypes() const noexcept override {
+    return {"Cast", "Transpose"};
```

Comment:
as we synced, let's remove Transpose since it already been opted by transpose opt.